src/ros_filter.cpp

```diff
@@ -3038,10 +3038,41 @@ namespace RobotLocalization
               (imuDynamicCorrectionData_[topicName].last_speed_ > imuDynamicCorrectionData_[topicName].min_speed_))
             {
               // Should be subtracted (-) since the offset is a difference.
-              imuDynamicCorrectionData_[topicName].yaw_offset_ = imuDynamicCorrectionData_[topicName].last_yaw_estimate_ - yaw;
-              // Should be added (+) since these are variances
-              imuDynamicCorrectionData_[topicName].yaw_offset_variance_ =
-                measurementCovariance(POSE_SIZE - 1, POSE_SIZE - 1) + imuDynamicCorrectionData_[topicName].last_yaw_variance_;
+              // The alpha-beta filter is rather special here, as it has to account for a few issues. First, how do we initialize it?
+              //  If initialized to zero, that means the yaw offset is assumed to be zero to start, with the resulting slow adjustments to
+              //  the filtered correction. The problem is that it's not zero. Without a better set of information (e.g. stored data from
+              //  a previous run), the best way is to initialize it with a jump to the first calculation, then run the filter.
+              // Because the alpha-beta filter is maintaining history, angle wrapping is a problem if not handled.
+              //  To correct this, we will always keep the offset in the range [-PI, PI]. The actual yaw angle
+              //  wrapping is handled by the Kalman filter. The offset angle wrapping needs to be handled here. The additional step is what
+              //  happens when the angle steps over the boundary (e.g. from -(PI-0.0001) to (PI-0.0001)). That case also has to be handled.
+              //  We could shift the old yaw offset to outside the boundary on the same side as the new yaw offset. That's a bit of work
+              //  for a corner case, so we will instead just skip the alpha-beta filter under those conditions, then running the
+              //  alpha-beta filter again.
+              double yaw_offset = FilterUtilities::clampRotation(imuDynamicCorrectionData_[topicName].last_yaw_estimate_ - yaw);
```

Comment:
The comments indicate that this line should be handling wrapping. The name of the function `clampRotation()` indicates that it is restricted to a range (I haven't seen the source or read the documentation for the function). Thinking through what `std::clamp()` does, this would be different from wrapping, so I just want to double-check what's going on here.

Reply:
The FilterUtilities::clampRotation function actually wraps. The code is basically:
while(number > PI)
{
number -= 2*PI;
}
while(number < -PI)
{
number += 2*PI
}